stubs/Pillow/PIL/Image.pyi

```diff
@@ -22,7 +22,7 @@ _Box: TypeAlias = tuple[int, int, int, int]
 _ConversionMatrix: TypeAlias = Union[
     tuple[float, float, float, float], tuple[float, float, float, float, float, float, float, float, float, float, float, float],
 ]
-_Color: TypeAlias = float | tuple[float, ...]
+_Color: TypeAlias = Union[int, tuple[int], tuple[int, int, int], tuple[int, int, int, int], str]
```

Comment:
This isn't correct. For some image modes, a `float` value will do. Here's an example using the "F" mode for both `.new()` and `.putpixel()`:

```python
>>> from PIL import Image
>>> x = Image.new('F', (100, 100), 34.98)
>>> x.putpixel((98, 97), 76.32)
>>> x.show()
```

We could be fancy with overloads and generics, but for now I suggest doing this:

```suggestion
# `str` values are only accepted if mode="RGB" for an `Image` object
# `float` values are only accepted for certain modes such as "F"
# See https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.new
_Color: TypeAlias = Union[int, tuple[int], tuple[int, int, int], tuple[int, int, int, int], str, float]
```

Reply:
Good catch. Overloaded generics sounds complicated and error-prone-- let's not do that.

In my brief testing, you can do a 1-tuple for a floating-point color, (but not any longer than that), so we should add that as well.

```python
>>> from PIL import Image
>>> Image.new('F', (100, 100), (34.98,))
```

I'm still not too familiar with code reviews; can I make a change to your suggestion before committing, or do I need to make two commits?

Reply:
It's easiest to make two commits.

Reply:
Yeah, either works -- either commit my suggestion and then do a follow-up commit, or disregard my suggestion and make your own commit. It'll all be squash-merged in the end anyway, so don't worry about making a mess with the commit history :)